### ISSUE TITLE
Sidestep Protobuf-java binary incompatibility

### ISSUE THREAD

**Issue by viktorklang** [akka/akka#13783]
imported from https://www.assembla.com/spaces/akka/tickets/3783

Best idea so far is embedding the protobuf-java-2.4.1 (current version used in Akka) under a different package name into akka-remote.

http://stackoverflow.com/questions/13746737/hand-edit-a-jar-to-change-package-names

akka.com.google... ?

**Comment by rkuhn** [comment]
This is the renaissance ;-) Currently looking into just including the (patched) sources in a different package.

**Comment by clockfly** [comment]
Almost 2 years. Are there plans to fix this in recent release?

I saw there was a PR, and the PR was closed due to disagreement. 

IMHO, it is acceptable for ProtobufSerializer to still depends on google.protobuf  as an exception, as long as user app can choose to exclude protobuf.jar manually when linking akka. Maybe the code is not clean, but it can solve the protobuf version conflict issue. 

It is not good to sleep this issue. Hadoop and other infrastructure services depends on protobuf. Allowing this issue to still exist will make user app extremely difficult to program in akka to connect to these services.
